### PR TITLE
test: Cypress - Added cypress grep library 

### PR DESCRIPTION
**Pull request by sharanya-appsmith**
Leveraging the library
[cypress-grep](https://github.com/cypress-io/cypress/tree/develop/npm/grep).
Using this we can tag testcases with relevant tags and use it to run
specific testcases.

**Command to run in local:** 
`CYPRESS_grepTags=@tag.Binding,@tag.Git npx cypress run
`
Pass the tags to CYPRESS_grepTags argument and only the test cases which has the
tags passed will be picked to run. ex `@tag.Binding and @tag.Git` are the tag names here.

**Tags can be added in the description on the test case like** 
`{ tags: ["@tag.Datasource"] }` for a single tag
`{ tags: ["@tag.Datasource", "@tag.Git"] }` for multiple tags 

**How to run In CI**
Single Tag - `/ok-to-test tags=@tag.Binding`
Multiple tag - `/ok-to-test tags=@tag.Binding,@tag.Git`

**TODOs in the next release:** 
- [ ] Add tags.ts file with all needed tags
- [ ] Add tags to remaining spec files. 
- [ ] Fail the PR run if tags added are not from tag.ts and post the message on the same

#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update

#### How Has This Been Tested?
- [x] Manual
- [ ] JUnit
- [ ] Jest
- [x] Cypress

#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [x] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [x] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
  - Added a search functionality to the app.

- **Enhancements**
  - Integrated search bar at the top of the `Hero` component and a `Search` component to the `App` component.
  - Added styles for the search bar.

### CODE REVIEW

**Review by mohanarpit — APPROVED** [review]

**Review by mohanarpit — APPROVED** [review]

File: app/client/cypress/plugins/index.js [hunk 1]
```diff
@@ -223,6 +223,17 @@ module.exports = async (on, config) => {
     },
   });
 
+  console.log("Type of 'config.specPattern':", typeof config.specPattern);
```
[the remark this answers is not in the document]
Comment (sharanya-appsmith) [reply app/client/cypress/plugins/index.js:226]: info:  Added this for debugging purpose in future as config.specPattern is string[] and while using grep plugin is an object.

File: app/client/cypress/plugins/index.js [hunk 2]
```diff
@@ -223,6 +223,17 @@ module.exports = async (on, config) => {
...
+  /**
+   * Cypress grep plug return specPattern as object and with absolute path
+   */
+  if(typeof config.specPattern == 'object'){
+  config.specPattern = config.specPattern.map((spec) => {
+    return spec.replace(process.cwd() + "/", "");
+  });
+}
+console.log("config.specPattern:", config.specPattern);
```
Comment (mohanarpit) [inline app/client/cypress/plugins/index.js:235]: @sharanya-appsmith Nit: Can we please format the code properly to ensure it's consistent with the rest of the code base?

Comment (sharanya-appsmith) [reply app/client/cypress/plugins/index.js]: Done.

**Review by mohanarpit — APPROVED** [review]

**Pull request merged**